hawc/apps/animal/api.py

```diff
@@ -257,6 +259,22 @@ def rob_filter(self, request):
         serializer = self.get_serializer(qs, many=True)
         return Response(serializer.data)
 
+    @action(detail=False, methods=("post",))
+    def update_terms(self, request):
+        # data must have one or more endpoints
+        if len(request.data) == 0:
+            raise DRFValidationError("List of endpoints must be > 1")
+        # check assessment level permissions
+        endpoint = self.model.objects.get(pk=request.data[0].get("id"))
```

Comment:
This permissions check could be abused. You could pass ids from multiple assessments and just the first id on the list would be one you'd have permission to view.  It's probably better to pass the `assessment_id` and use the `get_assessment_id_param, and then do a permissions check.

Later in the update method we can validate that all endpoint ids are from this assessment (by passing the assessment into the update_terms method), but that's a  deeper check than what's done at the api level.

Reply:
Part of the validation in `update_terms` makes sure all of the endpoints are in the same assessment, so just checking permissions on one of the endpoints should be good enough.

Reply:
I agree with you that that check already exists in effect, but it will be something that we will forget a year from now when we just look at the code. A little refactoring will be good to clearly show the separation of concerns - the api can check "can this user change this assessment",  the update method can check "are all ids in this assessment".